recipes/LibriSpeech/ASR/transducer/train.py

```diff
@@ -315,7 +327,12 @@ def dataio_prepare(hparams):
     @sb.utils.data_pipeline.takes("wav")
     @sb.utils.data_pipeline.provides("sig")
     def audio_pipeline(wav):
-        sig = sb.dataio.dataio.read_audio(wav)
+        if hparams["speed_perturb"]:
+            sig = sb.dataio.dataio.read_audio(wav)
+
+            sig = hparams["speed_perturb"](sig.unsqueeze(0)).squeeze(0)
```

Comment:
FWIW you mentioned slow training at some point so I am wondering if it could be related to the speed perturbation being performed here. Could be worth investigating (wondering if it's done on CPU or GPU, what happens wrt. DDP, wrt. PyTorch threading because of workers, etc.)

Reply:
Personally, I didn't notice a performance degradation performing this in the model's `forward` when in train mode.

Reply:
I would need to double check, but moving it to the forward seems to have made the end-of-epoch validation step much faster..